### DISCUSSION TITLE
How to update container version tag on a running application?

### DISCUSSION THREAD

**Discussion by BaileyMillerSSI** [epinio/epinio#2747] [Q&A]
I have an application that I am building in Github during a CD step and deploying to the Github package registry. I would like to implement a step where my CD connects to Epinio and tells it that the container image has been updated at which point it will roll the new version out to my instances. Is that something that is possible purely using the CLI or some other simplistic approach?

Desired flow:
Git Push -> CD starts -> CD calculates new version tag -> CD creates new image and uploads to ghcr -> CD logins to Epinio cli -> CD updates application to use the new version tag -> Epinio rolls out the new version

I attempted `epinio app update {name} --app-chart imageUrl={newImage}` but I got the error `Unable to change app chart of active application`

![image](https://github.com/epinio/epinio/assets/14271740/4c2dff4d-4d7b-41ed-ab34-ac84a4066aae)

**Comment by andreas-kupries** [comment]
Hi Bailey.

Given your talk about container images can I assume that you launched application using 
```
epinio push ... --container-image-url {app-workload-image} ...
```
instead of from sources ?

If yes, then re-pushing using the new workload image should do the trick.

Note that the `--app-chart` is __not__ the workload image. It is the helm chart we wrap around the image to deploy it.
Changing that is indeed not possible when the application is active.
I suspect that having it in the update command was a typo.
Actually, looking again at the proposed command that is very bogus, with `--appchart imageUrl=foo`.
The argument to `--app-chart` is the name of the chart to use, like `standard` (which is the builtin).
The option does not take any kind `key=value` as you attempted.

May I ask how you came to the belief that this would be the correct command ?

**Reply by BaileyMillerSSI** [reply]
I created the application from the UI like this:
![image](https://github.com/epinio/epinio/assets/14271740/ca1b19ee-e599-4ecc-9797-3d093e25b39c)

My general confusion is I cannot seem to find good Epinio cli / Epinio (in general) resources anywhere.

**Reply by BaileyMillerSSI** [reply]
`epinio push --container-image-url {image} -n {name}` seems to have done the trick thanks! Now to work on getting this into Github Actions. Are you familiar with any existing epinio actions in Github or will I need to just install the CLI in the action?

**Reply by andreas-kupries** [reply]
Our documentation is at https://docs.epinio.io/

There are no premade epinio actions for GHA.
Installing a binary and then logging in and pushing should be ok.

:warning: The Epinio team is currently refocusing on another project.
